### DISCUSSION TITLE
Generate Single View Trigger only if a field has a certain value

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#424] [Q&A] [answered]
### What product are you having troubles with?

Fast Data

### What Console version are you using?

12.2.1

### Description

I have a Single View with a base projection that contains the field `A`.
For each projection update (not only on the base projection), I want the Single View trigger to be generated only if the field `A` is null on the base projection.

Do you have an idea on how I can reach this result?

I am using the new projection-storer and single-view-trigger-generator services.

Thanks!

**Comment by ianFar96** [comment, marked as answer]
Hi @Lucascanna in this case you could use a custom function on the `identifier` in the `projectionChangesSchema.json` of the single-view-trigger-generator. 
Here's the doc for it https://docs.mia-platform.eu/docs/fast_data/configuration/config_maps/projection_changes_schema#custom-functions.

`projectionChangesSchema.json`:
```json
{
  "version": "X.Y.Z",
  "config": {
    "SINGLE_VIEW_NAME": {
      "paths": [
        {
          "path": [ "PROJECTION_2", "PROJECTION_1", "BASE_PROJECTION"],
          "identifier": "__fromFile__[myCustomFunction.js]"
        }
      ]
    }
  }
}
```

`myCustomFunction.js`:
```javascript
// note: this has to be an AsyncGenerator
module.exports = async function *myCustomFunction({ logger, dbMongo }, document) {
  if (document.A === null) {
    yield {
      IDENTIFIER_FIELD_1: document.FIELD_1,
      IDENTIFIER_FIELD_2: document.FIELD_2,
    }
  }
  // otherwise do not yield so no aggregation will be triggered
}
```

**Reply by lucascanna** [reply]
Hi Ian,

Thanks for the reply. I am wondering if there is any difference between the solution you are suggesting (docs [here](https://docs.mia-platform.eu/docs/fast_data/configuration/config_maps/projection_changes_schema#custom-functions)) with the "custom strategy" solution (docs [here](https://docs.mia-platform.eu/docs/fast_data/configuration/config_maps/kafka_projection_updates)).

**Reply by ianFar96** [reply]
Hi Luca, yes there is, the one I recommended to you is what we call semi-automatic strategies. This is because the strategy is resolved automatically until the base projection, the only thing you need to do is specify how to map the base projection to the Projection Change identifier of the Single View (example in JS above). Instead, with the custom strategies or manual strategies you need to specify for each projection you have the algorithm to retrieve a Projection Change identifier, in that case you won't be able to use any strategy automatism and as a consequence the projectionChangesSchema.json won't be used.
